File: sources/net.sf.j2s.core/test/dev/j2s/org/sun/awt/NullComponentPeer.js
Clazz.declarePackage ("sun.awt");
Clazz.load (["java.awt.peer.LightweightPeer", "$.PanelPeer"], "sun.awt.NullComponentPeer", ["java.lang.IllegalStateException", "$.UnsupportedOperationException", "java.awt.Dimension", "$.Insets", "$.Point", "$.Rectangle"], function () {
c$ = Clazz.declareType (sun.awt, "NullComponentPeer", null, [java.awt.peer.LightweightPeer, java.awt.peer.PanelPeer]);
Clazz.overrideMethod (c$, "isObscured", 
function () {
return false;
});
Clazz.overrideMethod (c$, "canDetermineObscurity", 
function () {
return false;
});
Clazz.overrideMethod (c$, "isFocusable", 
function () {
return false;
});
Clazz.overrideMethod (c$, "setVisible", 
function (b) {
}, "~B");
Clazz.defineMethod (c$, "show", 
function () {
});
Clazz.defineMethod (c$, "hide", 
function () {
});
Clazz.overrideMethod (c$, "setEnabled", 
function (b) {
}, "~B");
Clazz.defineMethod (c$, "enable", 
function () {
});
Clazz.defineMethod (c$, "disable", 
function () {
});
Clazz.overrideMethod (c$, "paint", 
function (g) {
}, "java.awt.Graphics");
Clazz.overrideMethod (c$, "repaint", 
function (tm, x, y, width, height) {
}, "~N,~N,~N,~N,~N");
Clazz.overrideMethod (c$, "print", 
function (g) {
}, "java.awt.Graphics");
Clazz.overrideMethod (c$, "setBounds", 
function (x, y, width, height, op) {
}, "~N,~N,~N,~N,~N");
Clazz.defineMethod (c$, "reshape", 
function (x, y, width, height) {
}, "~N,~N,~N,~N");
Clazz.overrideMethod (c$, "coalescePaintEvent", 
function (e) {
}, "java.awt.event.PaintEvent");
Clazz.defineMethod (c$, "handleEvent", 
function (e) {
return false;
}, "java.awt.Event");
Clazz.defineMethod (c$, "handleEvent", 
function (arg0) {
}, "java.awt.AWTEvent");
Clazz.overrideMethod (c$, "getPreferredSize", 
function () {
return  new java.awt.Dimension (1, 1);
});
Clazz.overrideMethod (c$, "getMinimumSize", 
function () {
return  new java.awt.Dimension (1, 1);
});
Clazz.overrideMethod (c$, "getToolkit", 
function () {
return null;
});
Clazz.overrideMethod (c$, "getColorModel", 
function () {
return null;
});
Clazz.overrideMethod (c$, "getGraphics", 
function () {
return null;
});
Clazz.overrideMethod (c$, "getGraphicsConfiguration", 
function () {
return null;
});
Clazz.overrideMethod (c$, "getFontMetrics", 
function (font) {
return null;
}, "java.awt.Font");
Clazz.overrideMethod (c$, "dispose", 
function () {
});
Clazz.overrideMethod (c$, "setForeground", 
function (c) {
}, "java.awt.Color");
Clazz.overrideMethod (c$, "setBackground", 
function (c) {
}, "java.awt.Color");
Clazz.overrideMethod (c$, "setFont", 
function (f) {
}, "java.awt.Font");
Clazz.overrideMethod (c$, "updateCursorImmediately", 
function () {
});
Clazz.defineMethod (c$, "setCursor", 
function (cursor) {
}, "java.awt.Cursor");
Clazz.overrideMethod (c$, "requestFocus", 
function (lightweightChild, temporary, focusedWindowChangeAllowed, time, cause) {
return false;
}, "java.awt.Component,~B,~B,~N,sun.awt.CausedFocusEvent.Cause");
Clazz.defineMethod (c$, "createImage", 
function (producer) {
return null;
}, "java.awt.image.ImageProducer");
Clazz.defineMethod (c$, "createImage", 
function (width, height) {
return null;
}, "~N,~N");
Clazz.overrideMethod (c$, "prepareImage", 
function (img, w, h, o) {
return false;
}, "java.awt.Image,~N,~N,java.awt.image.ImageObserver");
Clazz.overrideMethod (c$, "checkImage", 
function (img, w, h, o) {
return 0;
}, "java.awt.Image,~N,~N,java.awt.image.ImageObserver");
Clazz.defineMethod (c$, "preferredSize", 
function () {
return this.getPreferredSize ();
});
Clazz.defineMethod (c$, "minimumSize", 
function () {
return this.getMinimumSize ();
});
Clazz.overrideMethod (c$, "getLocationOnScreen", 
function () {
return  new java.awt.Point (0, 0);
});
Clazz.overrideMethod (c$, "getInsets", 
function () {
return this.insets ();
});
Clazz.overrideMethod (c$, "beginValidate", 
function () {
});
Clazz.overrideMethod (c$, "endValidate", 
function () {
});
Clazz.defineMethod (c$, "insets", 
function () {
return  new java.awt.Insets (0, 0, 0, 0);
});
Clazz.defineMethod (c$, "isPaintPending", 
function () {
return false;
});
Clazz.overrideMethod (c$, "handlesWheelScrolling", 
function () {
return false;
});
Clazz.overrideMethod (c$, "createVolatileImage", 
function (width, height) {
return null;
}, "~N,~N");
Clazz.overrideMethod (c$, "beginLayout", 
function () {
});
Clazz.overrideMethod (c$, "endLayout", 
function () {
});
Clazz.overrideMethod (c$, "getBackBuffer", 
function () {
throw  new IllegalStateException ("Page-flipping is not allowed on a lightweight component");
});
Clazz.overrideMethod (c$, "destroyBuffers", 
function () {
});
Clazz.overrideMethod (c$, "isReparentSupported", 
function () {
return false;
});
Clazz.overrideMethod (c$, "reparent", 
function (newNativeParent) {
throw  new UnsupportedOperationException ();
}, "java.awt.peer.ContainerPeer");
Clazz.defineMethod (c$, "restack", 
function () {
throw  new UnsupportedOperationException ();
});
Clazz.defineMethod (c$, "isRestackSupported", 
function () {
return false;
});
Clazz.overrideMethod (c$, "layout", 
function () {
});
Clazz.overrideMethod (c$, "getBounds", 
function () {
return  new java.awt.Rectangle (0, 0, 0, 0);
});
});

File: sources/net.sf.j2s.java.core/src/sun/awt/NullComponentPeer.java
/*
 * Some portions of this file have been modified by Robert Hanson hansonr.at.stolaf.edu 2012-2017
 * for use in SwingJS via transpilation into JavaScript using Java2Script.
 *
 * Copyright (c) 2000, 2008, Oracle and/or its affiliates. All rights reserved.
 * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
 *
 * This code is free software; you can redistribute it and/or modify it
 * under the terms of the GNU General Public License version 2 only, as
 * published by the Free Software Foundation.  Oracle designates this
 * particular file as subject to the "Classpath" exception as provided
 * by Oracle in the LICENSE file that accompanied this code.
 *
 * This code is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
 * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
 * version 2 for more details (a copy is included in the LICENSE file that
 * accompanied this code).
 *
 * You should have received a copy of the GNU General Public License version
 * 2 along with this work; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
 *
 * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
 * or visit www.oracle.com if you need additional information or have any
 * questions.
 */

package sun.awt;

import java.awt.Color;
import java.awt.Component;
import java.awt.Cursor;
import java.awt.Dimension;
import java.awt.Event;
import java.awt.Font;
import java.awt.FontMetrics;
import java.awt.Graphics;
import java.awt.GraphicsConfiguration;
import java.awt.Image;
import java.awt.Insets;
import java.awt.Point;
import java.awt.Rectangle;
import java.awt.event.PaintEvent;
import java.awt.image.ColorModel;
import java.awt.image.ImageObserver;
import java.awt.image.ImageProducer;
import java.awt.image.VolatileImage;
import java.awt.peer.ContainerPeer;
import java.awt.peer.LightweightPeer;
import java.awt.peer.PanelPeer;


/**
 * Implements the LightweightPeer interface for use in lightweight components
 * that have no native window associated with them.  This gets created by
 * default in Component so that Component and Container can be directly
 * extended to create useful components written entirely in java.  These
 * components must be hosted somewhere higher up in the component tree by a
 * native container (such as a Frame).
 *
 * This implementation provides no useful semantics and serves only as a
 * marker.  One could provide alternative implementations in java that do
 * something useful for some of the other peer interfaces to minimize the
 * native code.
 *
 * This was renamed from java.awt.LightweightPeer (a horrible and confusing
 * name) and moved from java.awt.Toolkit into sun.awt as a public class in
 * its own file.
 *
 * @author Timothy Prinzing
 * @author Michael Martak
 */

public class NullComponentPeer implements LightweightPeer,
    PanelPeer {

    @Override
		public boolean isObscured() {
        return false;
    }

    @Override
		public boolean canDetermineObscurity() {
        return false;
    }

    @Override
		public boolean isFocusable() {
        return false;
    }

    @Override
		public void setVisible(boolean b) {
    }

    public void show() {
    }

    public void hide() {
    }

    @Override
		public void setEnabled(boolean b) {
    }

    public void enable() {
    }

    public void disable() {
    }

    @Override
		public void paint(Graphics g) {
    }

    @Override
		public void repaint(long tm, int x, int y, int width, int height) {
    }

    @Override
		public void print(Graphics g) {
    }

    @Override
		public void setBounds(int x, int y, int width, int height, int op) {
    }

    public void reshape(int x, int y, int width, int height) {
    }

    @Override
		public void coalescePaintEvent(PaintEvent e) {
    }

    public boolean handleEvent(Event e) {
        return false;
    }

    @Override
		public void handleEvent(java.awt.AWTEvent arg0) {
    }

    @Override
		public Dimension getPreferredSize() {
        return new Dimension(1,1);
    }

    @Override
		public Dimension getMinimumSize() {
        return new Dimension(1,1);
    }

    @Override
		public java.awt.Toolkit getToolkit() {
        return null;
    }

    @Override
		public ColorModel getColorModel() {
        return null;
    }

    @Override
		public Graphics getGraphics() {
        return null;
    }

    @Override
		public GraphicsConfiguration getGraphicsConfiguration() {
        return null;
    }

    @Override
		public FontMetrics  getFontMetrics(Font font) {
        return null;
    }

    @Override
		public void dispose() {
    // no native code
    }

    @Override
		public void setForeground(Color c) {
    }

    @Override
		public void setBackground(Color c) {
    }

    @Override
		public void setFont(Font f) {
    }

    @Override
		public void updateCursorImmediately() {
    }

    public void setCursor(Cursor cursor) {
    }

    @Override
		public boolean requestFocus
        (Component lightweightChild, boolean temporary,
         boolean focusedWindowChangeAllowed, long time, CausedFocusEvent.Cause cause) {
        return false;
    }

    @Override
		public Image createImage(ImageProducer producer) {
        return null;
    }

    @Override
		public Image createImage(int width, int height) {
        return null;
    }

    @Override
		public boolean prepareImage(Image img, int w, int h, ImageObserver o) {
        return false;
    }

    @Override
		public int  checkImage(Image img, int w, int h, ImageObserver o) {
        return 0;
    }

    @Override
		public Point getLocationOnScreen() {
        return new Point(0,0);
    }

    @Override
		public Insets getInsets() {
        return insets();
    }

    @Override
		public void beginValidate() {
    }

    @Override
		public void endValidate() {
    }

    public Insets insets() {
        return new Insets(0, 0, 0, 0);
    }

    public boolean isPaintPending() {
        return false;
    }

    @Override
		public boolean handlesWheelScrolling() {
        return false;
    }

    @Override
		public VolatileImage createVolatileImage(int width, int height) {
        return null;
    }

    @Override
		public void beginLayout() {
    }

    @Override
		public void endLayout() {
    }

//    public void createBuffers(int numBuffers, BufferCapabilities caps)
//        throws AWTException {
//        throw new AWTException(
//            "Page-flipping is not allowed on a lightweight component");
//    }
    @Override
		public Image getBackBuffer() {
        throw new IllegalStateException(
            "Page-flipping is not allowed on a lightweight component");
    }
//    public void flip(int x1, int y1, int x2, int y2,
//                     BufferCapabilities.FlipContents flipAction)
//    {
//        throw new IllegalStateException(
//            "Page-flipping is not allowed on a lightweight component");
//    }
    @Override
		public void destroyBuffers() {
    }

    /**
     * @see java.awt.peer.ComponentPeer#isReparentSupported
     */
    @Override
		public boolean isReparentSupported() {
        return false;
    }

    /**
     * @see java.awt.peer.ComponentPeer#reparent
     */
    @Override
		public void reparent(ContainerPeer newNativeParent) {
        throw new UnsupportedOperationException();
    }

    /**
     * @see java.awt.peer.ContainerPeer#restack
     */
    public void restack() {
        throw new UnsupportedOperationException();
    }

    /**
     * @see java.awt.peer.ContainerPeer#isRestackSupported
     */
    public boolean isRestackSupported() {
        return false;
    }
    @Override
		public void layout() {
    }

    @Override
		public Rectangle getBounds() {
        return new Rectangle(0, 0, 0, 0);
    }


//    /**
//      * Applies the shape to the native component window.
//      * @since 1.7
//      */
//    public void applyShape(Region shape) {
//    }
}
